Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -245,56 +247,33 @@ void MediaSource::completeSeek()
     auto seekTarget = *m_pendingSeekTarget;
     m_pendingSeekTarget.reset();
 
-    struct SeeksCallbackAggregator final : public RefCounted<SeeksCallbackAggregator> {
-        SeeksCallbackAggregator(MediaTime target, MediaSource& source, CompletionHandler<void(const MediaTime&)>&& completionHandler)
-            : time(target)
-            , mediaSource(source)
-            , completionHandler(WTFMove(completionHandler))
-        {
-            ASSERT(this->completionHandler);
+    Ref<MediaTimePromise> promise = SourceBuffer::ComputeSeekPromise::all(RunLoop::main(), WTF::map(*m_activeSourceBuffers, [&](auto&& sourceBuffer) {
```

Comment:
We usually only use RunLoop::main() on WK2 code since there is the web thread on WK1.
RunLoop::current() might be better.
Ditto elsewhere probably

Reply:
The aim here will be to set the thread where we will run as a class member. Right now it's the main thread. May E I should use WorkQueue::main() instead

Reply:
done

Reply:
FYI: `RunLoop::main()` will be the main thread, even on iOS WK1. `WorkQueue::main()` will also be the main thread.  `callOnMainThread()` will run on the Web Thread (not the main thread, or the UI thread).